### PR TITLE
Implement as much of opaque framebuffers as we can without surfman

### PR DESCRIPTION
**Pull request by asajeffrey**
This PR implements as much of opaque framebuffers as we can without surfman. Once surfman lands, opaque framebuffers can be backed by surfman swap chains rather than shared textures.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24326)

### CODE REVIEW

File: components/script/dom/xrsession.rs [hunk 1]
```diff
@@ -73,6 +73,9 @@ pub struct XRSession {
     end_promises: DomRefCell<Vec<Rc<Promise>>>,
     /// https://immersive-web.github.io/webxr/#ended
     ended: Cell<bool>,
+    /// Opaque framebuffers need to know the session is "outside of a requestAnimationFrame"
+    /// https://immersive-web.github.io/webxr/#opaque-framebuffer
+    outside_raf: Cell<bool>,
```
Comment (Manishearth) [inline components/script/dom/xrsession.rs]: kinda would prefer this boolean to be inverted, but no biggie

**Review by Manishearth — APPROVED** [review]
r=me with or without change

File: components/script/dom/xrsession.rs (hunk 1 above)
Comment (asajeffrey) [reply components/script/dom/xrsession.rs]: I was following the way round it is in the spec, which is just as backwards!

Comment (Manishearth) [reply components/script/dom/xrsession.rs]: works for me

**Comment by jdm** [comment]
@asajeffrey Is this still necessary?

**Comment by asajeffrey** [comment]
No, this all got rolled into servo+surfman.

**Pull request closed without merging**